src/averageAges.js

```diff
@@ -55,7 +70,45 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childsList = !onlyWithSon
+    ? getChildsList(people)
+    : getChildsList(people).filter(item => item.sex === 'm');
+
+  const getMotherList = getMother(people);
+
+  const motherAge
+    = childsList.map(item => item.born
+      - getMotherList.find(person => item.mother === person.name).born);
+
+  return calclulationSumAndAverage(motherAge);
+}
+
+function calclulationSumAndAverage(ageList) {
```

Comment:
```suggestion
function getAverageValue(ageList) {
```
ты возвращаешь из функции только среднее значение, поэтому нет смысла указывать sum в названии переменной